src/app/core/language/language.module.ts

```diff
@@ -12,7 +14,12 @@ import { LanguageService } from "./language.service";
  */
 @NgModule({})
 export class LanguageModule {
-  constructor(translationService: LanguageService) {
+  constructor(
+    translationService: LanguageService,
+    enumService: ConfigurableEnumService,
+  ) {
     translationService.initDefaultLanguage();
+    // Making locales enum available at runtime
+    enumService["cacheEnum"](availableLocales);
```

Comment:
I have refactored the locales to be a `ConfigurableEnum`. 
This allows us to use the existing components for selecting but some things like adding new enum values don't really make sense and could lead to problems.
The alternative would be to have a new data type with explicit but not configurable values (just a enum).